files/lib/data/conversation/ConversationAction.class.php

```diff
@@ -216,6 +216,11 @@ public function markAsRead() {
 		if (empty($this->parameters['visitTime'])) {
 			$this->parameters['visitTime'] = TIME_NOW;
 		}
+
+		// userID cannot be set during an AJAX request, but only when calling this method within PHP
```

Comment:
Changing the wording does not fix anything, the actual issue is that the user id _can_ be present and you'll need to account for that. Your current code runs under the assumption, that the user id can never be provided with the request, when in fact it can.

In fact, you do need to update your code to address this _and_ adjust the comment.